### ISSUE TITLE
2.0rc hapi integration - runtime error when gui: true

### ISSUE THREAD

**Issue by SkaterDad** [apollographql/apollo-server#1195]
When trying to add Apollo Server 2.0rc to my existing hapi project last night, I ran into a runtime error when attempting to use the GUI.  I was able to POST requests just fine, but could not open GraphQL Playground.

I wanted to debug it a bit last night, but the transpiled code is very hard to follow.  It would be so much easier if you didn't have to support Node 6 anymore, or could ship a build that's usable natively for Node 8+.

Minimal code to reproduce ([Also on Glitch](https://glitch.com/edit/#!/hollow-hour))
```js

var Hapi = require('hapi')

var { ApolloServer, gql } = require('apollo-server-hapi')

var server = new Hapi.Server({ port: process.env.PORT })


var typeDefs = gql`
  type Query {
    hello: String
  }
`

var resolvers = {
  Query: {
    hello: () => "Test"
  }
}

var apollo = new ApolloServer({ typeDefs, resolvers })
apollo.applyMiddleware({
  app: server,
  gui: true,
  path: '/' 
})

server.start().then(() => console.log('running'))
```

```
Debug: internal, implementation, error 

    TypeError: Cannot read property 'subscriptionsPath' of null

    at /rbd/pnpm-volume/485b790f-ae73-40af-a553-dae4f20dd9bd/node_modules/.registry.npmjs.org/apollo-server-hapi/2.0.0-rc.0/node_modules/apollo-server-hapi/dist/ApolloServer.js:67:118

    at Generator.next (<anonymous>)

    at fulfilled (/rbd/pnpm-volume/485b790f-ae73-40af-a553-dae4f20dd9bd/node_modules/.registry.npmjs.org/apollo-server-hapi/2.0.0-rc.0/node_modules/apollo-server-hapi/dist/ApolloServer.js:4:58)

    at <anonymous>

    at process._tickCallback (internal/process/next_tick.js:188:7)
```

**Comment by evans** [comment]
Ah thank you for reporting the issue! I'm taking a look and will add some of the tests that we are using for the express integration

**Comment by evans** [comment]
Also thank you so much for the reproduction! See https://glitch.com/edit/#!/hapi-as2?path=server.js:13:17 for a fork that upgrades to the latest RC

**Comment by SkaterDad** [comment]
@evans Thanks for the quick fix!  Apollo 2.0 is a really exciting project :+1: 

Would you be willing to add the ability to pass more hapi route options to the plugin/routes, not just `cors`?  hapi servers can be set up with default options that get applied to each route, and get overridden on a case-by-case basis.

There are quite a few route options:  https://hapijs.com/api#route-options

For my own cases, I can see `auth`, `cors`, and `cache` being useful/necessary, but other people may need more.

 GraphQL handles auth in such a different way, I need to provide a different `auth` config object to the endpoint.  My server's default auth strategy requires admin rights, but I would want the GraphQL endpoint to have optional auth that I can attach to the context object if present to do access control at the resolver level.

I can create a new issue, or post this in an existing issue if you prefer for tracking purposes.
